roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -268,138 +293,202 @@ def get_image_size(self):
         return self.image_masks.shape[:2]
 
     @staticmethod
-    def write_segmentation(segext_obj, savepath, metadata_dict, **kwargs):
-        if isinstance(metadata_dict, str):
-            with open(metadata_dict, 'r') as f:
-                metadata_dict = yaml.safe_load(f)
-
-        # NWBfile:
-        nwbfile_args = dict(identifier=str(uuid.uuid4()), )
-        if 'NWBFile' in metadata_dict:
-            nwbfile_args.update(**metadata_dict['NWBFile'])
-        nwbfile = NWBFile(**nwbfile_args)
-
-        # Subject:
-        if 'Subject' in metadata_dict:
-            nwbfile.subject = Subject(**metadata_dict['Subject'])
-
-        # Device:
-        if isinstance(metadata_dict['ophys']['Device'], list):
-            for devices in metadata_dict['ophys']['Device']:
-                nwbfile.create_device(**devices)
+    def get_nwb_metadata(sgmextractor):
+        """
+        Converts metadata from the segmentation into nwb specific metadata
+        Parameters
+        ----------
+        sgmextractor: SegmentationExtractor
+        """
+        metadata = {'NWBFile': {'session_start_time': datetime.now(),
+                                'identifier': str(uuid.uuid4()),
+                                'session_description': 'ses desc'},
+                    'ophys': {'Device': [{'name': 'Microscope'}],
+                              'Fluorescence': {'roi_response_series':[{'name': 'RoiResponseSeries',
+                                                                       'description': 'array of raw fluorescence traces'}]},
+                              'ImageSegmentation': {'plane_segmentations': [{'description': 'Segmented ROIs',
+                                                                            'name': 'PlaneSegmentation'}]},
+                              'ImagingPlane':[{'name': 'ImagingPlane',
+                                               'description': 'imaging plane description',
+                                               'excitation_lambda': 600.0,
+                                               'indicator': 'GCaMP',
+                                               'optical_channels': [{'name': 'OpticalChannel',
+                                                                     'emission_lambda': 500.0,
+                                                                     'description': 'description of optical channel'}]}],
+                              'TwoPhotonSeries': [{'name': 'TwoPhotonSeries'}]}}
+        # Optical Channel name:
+        for i in range(sgmextractor.get_num_channels()):
+            ch_name = sgmextractor.get_channel_names()[i]
+            if i == 0:
+                metadata['ophys']['ImagingPlane'][0]['optical_channels'][i]['name'] = ch_name
+            else:
+                metadata['ophys']['ImagingPlane'][0]['optical_channels'].append(dict(
+                    name=ch_name,
+                    emission_lambda=500.0,
```

Comment:
```suggestion
                    emission_lambda=np.nan,
```
please don't make up data!

Reply:
yes, thanks for catching that, fixed